src/Styleguide/Components/Stepper.tsx

```diff
@@ -0,0 +1,86 @@
+import { color, space } from "@artsy/palette"
+import { CheckIcon } from "Assets/Icons/CheckIcon"
+import React from "react"
+import styled from "styled-components"
+import { Tab, Tabs, TabsProps } from "Styleguide/Components/Tabs"
+import { Flex } from "Styleguide/Elements/Flex"
+
+interface StepperProps extends TabsProps {
+  initialTabIndex: number
+
+  /* the step user currently is at (e.g. previous steps completed) */
+  currentStepIndex: number
+
+  disableNavigation?: boolean
+}
+
+const transformTabBtn = (
+  element: JSX.Element,
+  tabIndex: number,
+  props: any
+): JSX.Element => {
+  const { currentStepIndex, disableNavigation } = props
+  const elementWithoutNav = React.cloneElement(element, {
+    ...element.props,
+    onClick: () => {
+      null
+    },
+  })
+  if (tabIndex > currentStepIndex) {
+    // don't allow users to jump ahead
+
+    return elementWithoutNav
+  } else if (currentStepIndex && tabIndex < currentStepIndex) {
+    return (
+      <Flex>
+        <CheckMarkWrapper>
+          <CheckIcon color={color("green100")} />
+        </CheckMarkWrapper>
+        {disableNavigation ? elementWithoutNav : element}
+        <div /> {/* hack for getting rid of last-child in Tabs.tsx */}
```

Comment:
😅 So this doesn't result in malformed markup?

Reply:
Nope. Every tag in JSX can be self closing.

Reply:
Lol I read that as a closing tag, rather than self-closing 🙈